### PR TITLE
fix missing nack reports

### PR DESCRIPTION
**Pull request by davibe**
- register: clear statuses on rollover + seq jump
- fix typo

### CODE REVIEW

**Review by k0nserv — APPROVED** [review]
I don't fully understand, but anyway. Just one question

File: src/streams/register.rs [hunk 1]
```diff
@@ -108,13 +108,13 @@ impl ReceiverRegister {
         if did_wrap {
             // The indices wrapped around the end of `packet_status`, we clear any entries between
             // the current sequence number and nack_check_from.
-            let start = (*seq + 1).into();
+            let start = *seq - self.packet_index(*seq) as u64;
```
[the remark this answers is not in the document]
Comment (davibe) [reply src/streams/register.rs:111]: by definition of `%` I think

**Comment by davibe** [comment]
I can try to explain

```
F = check from
M = max seq
- = missing
x = received
N = newly received

Buffer

0: |xxxxxxxxFN--|
1: |xxNxxxxxxF--|
```

Previously, at this stage, `did_wrap` would be true, therefore we would clear out the state from N to F

```
2: |xxN------F--|
```

  Because of the seq jump, we would leave the initial positions dirty, and we'd consider them received (we would not nack those packets).

This problem gets masked when packets after F are received, because F would move forward and clear things out. However, this does not happen with loss.

Now we clear out from the beginning of the buffer on first rollover. 
```
|--N------F--|
```

**Comment by k0nserv** [comment]
Great find! Thanks for the explanation

**Pull request merged**